selfservice/flow/recovery/error.go

```diff
@@ -83,7 +83,10 @@ func (s *ErrorHandler) WriteFlowError(
 			return
 		}
 
-		if f.Type == flow.TypeAPI || x.IsJSONRequest(r) {
+		// We need to use the new flow, as that flow will be a browser flow. Bug fix for:
+		//
+		// https://github.com/ory/kratos/issues/2049
```

Comment:
```suggestion
		// https://github.com/ory/kratos/issues/2049!!
```

https://github.com/ory/closed-reference-notifier/commit/5d419e0124280696469e6e862a4ab1cfb39f9d84